pkg/asset/installconfig/vsphere/validation.go

```diff
@@ -66,3 +74,16 @@ func folderExists(ic *types.InstallConfig, fldPath *field.Path) field.ErrorList
 	}
 	return nil
 }
+
+// validateCPUConfig returns an error if a number of cores per socket (coresPerSocket) has lower value than number of CPUs (cpus).
+func validateCPUConfig(ic *types.InstallConfig) error {
+	if ic.Platform.VSphere.DefaultMachinePlatform != nil {
```

Comment:
You will want to validate the default, worker, and master machine pools, but if you move the validation to pkg types as I suggested in the other comment that should be handled for you.